core/domain/classifier_services.py

```diff
@@ -124,6 +124,35 @@ def classify_string_classifier_rule(state, normalized_answer):
 
     return None
 
+def check_classification_condition(state):
+    """Checks if conditions for training are satisfied.
+
+    Args:
+        state: Domain object for the State.
+
+    Returns:
+        bool: True, if the conditions are satisfied.
+    """
+    training_examples = [
+        [doc, []] for doc in state.interaction.confirmed_unclassified_answers]
+    labels_ctr = 0
+    for (answer_group_index, answer_group) in enumerate(
+            state.interaction.answer_groups):
+        classifier_rule_spec_index = answer_group.get_classifier_rule_index()
+        if classifier_rule_spec_index is not None:
+            classifier_rule_spec = answer_group.rule_specs[
+                classifier_rule_spec_index]
+        else:
+            classifier_rule_spec = None
+        if classifier_rule_spec is not None:
```

Comment:
I'm confused. Let's say you drop lines 145 to 147. From my reading, I think that this preserves the existing behaviour of the code. However, maybe I'm misunderstanding; what is the specific circumstance in which the behaviour would be different?